src/main/java/it/gov/pagopa/initiative/statistics/service/BaseKafkaConsumer.java

```diff
@@ -0,0 +1,59 @@
+package it.gov.pagopa.initiative.statistics.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import it.gov.pagopa.common.kafka.utils.KafkaConstants;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.Header;
+
+import java.nio.charset.StandardCharsets;
+
+@Slf4j
+public abstract class BaseKafkaConsumer<E> {
+    protected final String applicationName;
+    protected final String consumerGroup;
+    protected final ObjectReader objectReader;
+
+    protected BaseKafkaConsumer(String applicationName, String consumerGroup, ObjectMapper objectMapper) {
+        this.applicationName = applicationName;
+        this.consumerGroup = consumerGroup;
+        this.objectReader = objectMapper.readerFor(getRecordClass());
+    }
+
+    protected abstract Class<E> getRecordClass();
+
+
+    /** It will check if the current record is not a RETRY of another application */
+    protected boolean isNotRetry(Pair<ConsumerRecord<String, String>, E> r2e) {
```

Comment:
this signature is not as generic as desired for a class supposed to serve each Consumer, inside this method we are using just the ConsumerRecord, let's accept just it instead of the Pair, which will force you to create a Pair to use it
in the BaseStatisticsEvaluationService class you could create the method accepting the Pair, which will call this